bug(misconf): trivy image panics on null Config.Healthcheck with bare HEALTHCHECK history line

## Description

Trivy panics when scanning an image whose history contains a bare `HEALTHCHECK {...}` line and `Config.Healthcheck` is null.

This combination doesn't occur from a single docker build — docker always materializes `HEALTHCHECK NONE` as non-null `Healthcheck{Test:["NONE"]}`. It happens when an image built with docker/BuildKit is re-committed with buildah, which normalizes `Test:["NONE"]` to null while leaving the history line untouched.

Dockerfile:
```dockerfile
FROM alpine:3.20
HEALTHCHECK NONE
CMD ["/bin/sh"]
```

Steps:
```bash
# 1. Build
docker build -t diss-10902:latest -f Dockerfile .

# 2. Re-commit with buildah
ctr="$(buildah from docker-daemon:diss-10902:latest)"
img="$(buildah commit -q "$ctr")"
buildah rm "$ctr"
rm -f out.tar
buildah push "$img" docker-archive:./out.tar:diss-10902-patched:latest

# 3. Load
docker load -i out.tar

docker inspect diss-10902-patched:latest --format '{{json .Config.Healthcheck}}'
# -> null

docker history diss-10902-patched:latest --no-trunc --format '{{json .}}' | jq -r 'select(.CreatedBy | contains("HEALTHCHECK")) | .CreatedBy'
# -> HEALTHCHECK {Test:[NONE] Interval:0s Timeout:0s StartPeriod:0s StartInterval:0s Retries:0}

# 4. Scan

## non-patched
trivy image --image-config-scanners misconfig diss-10902:latest --scanners "" -q

diss-10902:latest (dockerfile)

Tests: 27 (SUCCESSES: 26, FAILURES: 1)

## patched
trivy image --image-config-scanners misconfig diss-10902-patched:latest --scanners "" -q
panic: runtime error: invalid memory address or nil pointer dereference
```

## Expected
Trivy scans the image without crashing.

## Actual
Trivy panics.

### Discussed in https://github.com/aquasecurity/trivy/discussions/10902

--- a/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile.go
+++ b/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile.go
@@ -147,6 +147,15 @@ func buildRunInstruction(s string) string {
 }
 
 func buildHealthcheckInstruction(health *v1.HealthConfig) string {
+	// A history entry's CreatedBy can start with "HEALTHCHECK" while the
+	// resolved config carries no Healthcheck object — e.g. `HEALTHCHECK NONE`,
+	// or images built by buildah/buildkit that record a HEALTHCHECK history
+	// line without materializing cfg.Config.Healthcheck. Dereferencing the nil
+	// pointer below would crash the image-config (misconfig) scanner, so treat
+	// the absent config as the disabled form.
+	if health == nil {
+		return "HEALTHCHECK NONE"
+	}
 	var interval, timeout, startPeriod, retries, command string
 	if health.Interval != 0 {
 		interval = fmt.Sprintf("--interval=%s ", health.Interval)

--- a/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile.go
+++ b/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile.go
@@ -147,12 +147,8 @@ func buildRunInstruction(s string) string {
 }
 
 func buildHealthcheckInstruction(health *v1.HealthConfig) string {
-	// A history entry's CreatedBy can start with "HEALTHCHECK" while the
-	// resolved config carries no Healthcheck object — e.g. `HEALTHCHECK NONE`,
-	// or images built by buildah/buildkit that record a HEALTHCHECK history
-	// line without materializing cfg.Config.Healthcheck. Dereferencing the nil
-	// pointer below would crash the image-config (misconfig) scanner, so treat
-	// the absent config as the disabled form.
+	// Config.Healthcheck can be null even when a HEALTHCHECK history line exists
+	// if the image was re-committed by a tool that normalizes HEALTHCHECK NONE to null.
 	if health == nil {
 		return "HEALTHCHECK NONE"
 	}

--- a/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile_test.go
+++ b/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile_test.go
@@ -459,6 +459,24 @@ ENTRYPOINT ["/bin/sh"]
 `,
 		},
 		{
+			// A HEALTHCHECK history line whose resolved config carries no
+			// Healthcheck object (Config.Healthcheck == nil). This happens with
+			// `HEALTHCHECK NONE` and with buildah/buildkit images that record a
+			// HEALTHCHECK history line without materializing the config object.
+			// Before the nil guard in buildHealthcheckInstruction this panicked
+			// the misconfig scanner with a nil pointer dereference.
+			name: "healthcheck history without resolved config",
+			input: &v1.ConfigFile{
+				History: []v1.History{
+					{
+						CreatedBy: "HEALTHCHECK NONE",
+					},
+				},
+				// Config.Healthcheck deliberately nil
+			},
+			expected: "HEALTHCHECK NONE\n",
+		},
+		{
 			name: "legacy env format",
 			input: &v1.ConfigFile{
 				History: []v1.History{

--- a/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile_test.go
+++ b/pkg/fanal/analyzer/imgconf/dockerfile/dockerfile_test.go
@@ -459,12 +459,6 @@ ENTRYPOINT ["/bin/sh"]
 `,
 		},
 		{
-			// A HEALTHCHECK history line whose resolved config carries no
-			// Healthcheck object (Config.Healthcheck == nil). This happens with
-			// `HEALTHCHECK NONE` and with buildah/buildkit images that record a
-			// HEALTHCHECK history line without materializing the config object.
-			// Before the nil guard in buildHealthcheckInstruction this panicked
-			// the misconfig scanner with a nil pointer dereference.
 			name: "healthcheck history without resolved config",
 			input: &v1.ConfigFile{
 				History: []v1.History{
@@ -472,7 +466,6 @@ ENTRYPOINT ["/bin/sh"]
 						CreatedBy: "HEALTHCHECK NONE",
 					},
 				},
-				// Config.Healthcheck deliberately nil
 			},
 			expected: "HEALTHCHECK NONE\n",
 		},

